server/util/compression/compression.go

```diff
@@ -138,6 +148,9 @@ func (d *zstdDecompressor) Close() error {
 // at least a few hundred bytes.
 func NewZstdCompressingReader(reader io.Reader, readBuf []byte, compressBuf []byte) (io.ReadCloser, error) {
```

Comment:
would something like this work ?

```
counter := ioutil.Counter{}
compressed := compression.NewZstdCompressingReader(io.TeeReader(counter, reader), ...)
doSomethingWith(counter.Count())
```

Reply:
Nice! Done. @bduffany could you look over the final version when you have a chance?